### ISSUE TITLE
apt-get install & upgrade failures cite Alexa as cause

### ISSUE THREAD

**Issue by dpenney5** [alexa/alexa-avs-sample-app#262]
Hi,
am installing a new package & upgrading. However process fails referring to Alexa as cause.
Would you please help because this has now halted my project.
I would remove Alexa but I can't see instructions on how to do so.
Please see output below...
Regards
## David

sudo apt-get install xbindkeys-config

Reading package lists... Done
Building dependency tree  
Reading state information... Done
The following packages were automatically installed and are no longer required:
  cracklib-runtime db-util db5.3-util libcrack2 libpam-cracklib libperl4-corelibs-perl lsof rc
Use 'apt-get autoremove' to remove them.
The following NEW packages will be installed:
  xbindkeys-config
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
Need to get 19.7 kB of archives.
After this operation, 79.9 kB of additional disk space will be used.
Get:1 http://mirrordirector.raspbian.org/raspbian/ jessie/main xbindkeys-config armhf 0.1.3-2 [19.7 kB]
Fetched 19.7 kB in 0s (24.2 kB/s)  
Selecting previously unselected package xbindkeys-config.
(Reading database ... 120545 files and directories currently installed.)
Preparing to unpack .../xbindkeys-config_0.1.3-2_armhf.deb ...
Unpacking xbindkeys-config (0.1.3-2) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up screen (4.2.1-3+deb8u1) ...
insserv: warning: script 'S99alexa' missing LSB tags and overrides
insserv: warning: script 'alexa' missing LSB tags and overrides
insserv: There is a loop at service alexa if started
insserv: There is a loop between service plymouth and mountnfs if started
insserv:  loop involving service mountnfs at depth 7
insserv:  loop involving service networking at depth 6
insserv: There is a loop between service plymouth and mountall-bootclean if started
insserv:  loop involving service mountall-bootclean at depth 6
insserv:  loop involving service mountall at depth 5
insserv: There is a loop between service alexa and checkroot if started
insserv:  loop involving service checkroot at depth 3
insserv:  loop involving service mountdevsubfs at depth 2
insserv: There is a loop at service plymouth if started
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Max recursions depth 99 reached
insserv:  loop involving service vncboot at depth 1
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv:  loop involving service urandom at depth 8
insserv:  loop involving service checkroot-bootclean at depth 5
insserv: There is a loop between service alexa and mountkernfs if started
insserv:  loop involving service mountkernfs at depth 1
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv:  loop involving service hdparm at depth 4
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: There is a loop between service plymouth and mountnfs-bootclean if started
insserv:  loop involving service mountnfs-bootclean at depth 8
insserv: Starting alexa depends on plymouth and therefore on system facility`$all' which can not be true!
insserv: Starting alexa depends on plymouth and therefore on system facility `$all' which can not be true!
insserv: exiting now without changing boot order!
update-rc.d: error: insserv rejected the script header
dpkg: error processing package screen (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up xbindkeys-config (0.1.3-2) ...
Errors were encountered while processing:
 screen
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by somu1795** [comment]
Yes, this is a problem , and that's why I had to uninstall alexa.

**Comment by dpenney5** [comment]
Hi Somu1795

what did you use to uninstall Alexa?
Regards
David

**Comment by shuntera** [comment]
I have the same question. I want to get rid of Alex so that I can perform my updates and then add it back later - so other than the directory it lives in what else is running and needs to be gotten rid of.

**Comment by dpenney5** [comment]
No it's only running the alexapi app, take a look in the alexapi setup.sh
It uses init.d to run the S99alexa script, I think in rc5

If you go to that folder there are very clear instructions on how to get the system to remove it from the statup process. I don't have a Pi with me (I am travelling) - I will be back at the weekend and will look then.

But it's not too hard to figure out if you can read simple scripts starting with the setup.sh and  find the directory where it put S99alexa and look in the README file there and follow instructions on how to rename it to K99alexa then run the right script to remove it. After that you can delete it.
Sorry I can't be more specific right now.

Let me know how far you get and if you haven't got it done by the weekend, I will reply again with detailed instructions.

Regards
David

**Comment by dpenney5** [comment]
Hi Shuntera,
I looked in my Pi & here are clearer instructions if you need them:

The AlexaPi/setup.sh script puts a script alexa into the directory /etc/init.d then creates a link file called S99Alexa in the directory /etc/rc5.d that points to the /etc/init.d/alexa script plus creates a log file in /var/log/alexa.log

if you cd /etc/rc5.d and type the REAME file with the command cat README|more
then you will see how to remove the entry…

rename S99Alexa to K99Alexa then run the script ‘update-rc.d script defaults’ like this…

sudo cd /etc/rc5.d
sudo mv S99Alexa K99Alexa
sudo update-rc.d script defaults

You can now delete the Alexa link file in /etc/rc5.d and the script itself from /etc/init.d

sudo reboot now

and you should no longer have the encumbrance of this script getting in the way of upgrades

Myself I run alexa from within a script that runs as a cron job once per minute - checking to see if alexa is already running & if not then restarting it with output being recorded into a log file.

That way it is always in my control & not subject to any system upgrades that may occur.

Hope that helps.
Regards
David

**Comment by ajotwani** [comment]
We're closing this Issue since the original poster (or another commenter) hasn't yet responded to the question or request made to them 14 days ago. We therefore assume that the user has lost interest or resolved the problem on their own.

Don't worry though; if this is an error, let us know with a comment and we'll be happy to reopen the Issue.

Thanks!

**Comment by Sztucznooki** [comment]
this is still a problem

**Comment by amolpagar** [comment]
As David mentioned below steps worked fine

"No it's only running the alexapi app, take a look in the alexapi setup.sh
It uses init.d to run the S99alexa script, I think in rc5

If you go to that folder there are very clear instructions on how to get the system to remove it from the statup process. I don't have a Pi with me (I am travelling) - I will be back at the weekend and will look then.

But it's not too hard to figure out if you can read simple scripts starting with the setup.sh and find the directory where it put S99alexa and look in the README file there and follow instructions on how to rename it to K99alexa then run the right script to remove it. After that you can delete it.
Sorry I can't be more specific right now.

Let me know how far you get and if you haven't got it done by the weekend, I will reply again with detailed instructions."

but there is still a problem with update-rc.d script defaults command
unable to read /etc/init.d./script

looking into it

**Comment by amolpagar** [comment]
after digging more into this, looks like script, scanbd folders are not present in init.d directory.
basically this directory contains services which gets executed during boot I believe, there are ways to stop or disable these services
http://www.stuffaboutcode.com/2012/06/raspberry-pi-run-program-at-start-up.html 

finally my command was 
sudo update-rc.d alexa

hope it helps someone out there

**Comment by Anchises68** [comment]
David you're my hero today.  I was sweating this one.  I had almost identical problems.
I finally had finished setting up alexa with this git on my raspi when I needed to to make some changes/updates to my audio I couldn't run any apt-get without getting errors.

I also had problems with logging  into the shell.  I run vnc - see screen shot below for the errors I kept getting every time I ran a command. 
<img width="878" alt="screen shot 2017-12-14 at 2 15 50 pm" src="https://user-images.githubusercontent.com/6642959/34017156-6492ed4a-e0d9-11e7-840b-e52c176d064c.png">

I eventually removed everything alexa and followed David's example.  I'd like to start over with this git, but I'm concerned of running into the same problem.  Suggestions?
